static/js/shared/data_fetcher.ts

```diff
@@ -244,20 +251,28 @@ function getStatsDataFromCachedData(
   statsVars: string[],
   perCapita = false,
   scaling = 1,
-  cachedData: CachedStatsVarDataMap = {}
+  denominators: string[] = [],
+  cachedData: CachedStatVarDataMap = {}
 ): StatsData {
   const result = new StatsData(places, statsVars, [], {});
   const dates: Set<string> = new Set();
-  for (const place of places) {
-    for (const sv of statsVars) {
-      result.data[sv] = statApiResponse(place, cachedData[place][sv]);
-      if (perCapita) {
-        result.data[sv] = computePerCapita(
-          result.data[sv],
-          statApiResponse(place, cachedData[place][TOTAL_POPULATION_SV]),
-          scaling
-        );
-      }
+  for (let i = 0; i < statsVars.length; i++) {
+    const sv = statsVars[i];
+    result.data[sv] = statApiResponseFromCacheData(sv, places, cachedData);
+    if (perCapita) {
+      result.data[sv] = computePerCapita(
+        result.data[sv],
+        statApiResponseFromCacheData(TOTAL_POPULATION_SV, places, cachedData),
+        scaling
+      );
+    } else if (denominators.length) {
```

Comment:
might be helpful to check the denominators length is same as the base stat vars here or earlier

Reply:
done. since we don't have assertion support, printed an error message to the console.